osu.Game/Screens/OnlinePlay/Playlists/CreatePlaylistsRoomButton.cs

```diff
@@ -6,13 +6,11 @@
 
 namespace osu.Game.Screens.OnlinePlay.Playlists
 {
-    public class CreatePlaylistsRoomButton : PurpleTriangleButton
+    public class CreatePlaylistsRoomButton : CreateRoomButton
```

Comment:
"might as well" becomes quite a large change. It looks like moving all the handling to `MatchSettingsOverlay` may work best, rather than having it at individual buttons. Will look at that.

Reply:
I've made the changes required to make this work, but it's quite ugly. @smoogipoo when reviewing this I'm pretty sure you're going to find the current code quite convoluted, but it feels like some further (more involved) reorganisation would be required to avoid the redirection of loading states.

Also playlists support is not tested since the existing tests don't really lend to it, and adding proper testing is more involved.

If these points are seen as deal-breakers it may be best to revert this to only handle multiplayer until a later date..